Lib/textwrap.py

```diff
@@ -432,15 +432,28 @@ def dedent(text):
     lines = text.split('\n')
 
     # Get length of leading whitespace, inspired by ``os.path.commonprefix()``.
-    non_blank_lines = [l for l in lines if l and not l.isspace()]
-    l1 = min(non_blank_lines, default='')
-    l2 = max(non_blank_lines, default='')
-    margin = 0
+    l1 = None
```

Comment:
```suggestion
    l1 = ''
```
You can then skip the is None check in the loop. Similar for l2 (but you have to pick the default with care)